components/emoji_picker/components/emoji_picker_item.jsx

```diff
@@ -45,18 +53,23 @@ export default class EmojiPickerItem extends React.PureComponent {
         const {emoji} = this.props;
 
         if (emoji.category && emoji.batch) {
-            let className = 'emojisprite';
-
-            className += ' emoji-category-' + emoji.category + '-' + emoji.batch;
-            className += ' emoji-' + emoji.filename;
+            let itemClassName = 'emoji-picker__item';
+            itemClassName += this.props.isSelected ? ' selected' : '';
+            let spriteClassName = 'emojisprite';
+            spriteClassName += ' emoji-category-' + emoji.category + '-' + emoji.batch;
+            spriteClassName += ' emoji-' + emoji.filename;
 
             item = (
-                <div className={'emoji-picker__item'}>
+                <div
+                    className={itemClassName}
+                    ref={(emojiItem) => {
```

Comment:
It is how the react docs suggest doing it. (https://reactjs.org/docs/refs-and-the-dom.html)

I always assumed you want to rebind it to the new physical dom object on every re-render. I have always followed this blindly, never really thought to move it out. I will test to ensure it still works, and update this.

Reply:
Btw, thanks for the feedback. I will try and take a look at/make these updates tonight.

Reply:
I think @hmhealey is worried about from a PureComponent perspective, [this post explains it a bit](https://medium.freecodecamp.org/why-arrow-functions-and-bind-in-reacts-render-are-problematic-f1c08b060e36). That said, I'm not sure if refs are taken into account for rendering or not, so it might be a non-issue in this case. The [React docs mention it](https://reactjs.org/docs/refs-and-the-dom.html#caveats) but only say that it will cause the ref callback to be called twice, and doesn't talk about whether it affects re-rendering of pure components